sdks/js/packages/core/react/components/organization/api-keys/index.tsx

```diff
@@ -0,0 +1,109 @@
+import { Flex, Text, EmptyState, Button } from '@raystack/apsara/v1';
+import styles from './styles.module.css';
+import keyIcon from '~/react/assets/key.svg';
+import { Image } from '@raystack/apsara';
+import { useFrontier } from '~/react/contexts/FrontierContext';
+import { DEFAULT_API_PLATFORM_APP_NAME } from '~/react/utils/constants';
+import { FrontierClientAPIPlatformOptions } from '~/shared/types';
+import { useMemo } from 'react';
+import { PERMISSIONS, shouldShowComponent } from '~/utils';
+import { usePermissions } from '~/react/hooks/usePermissions';
+import { ExclamationTriangleIcon } from '@radix-ui/react-icons';
+
+const NoServiceAccounts = ({
+  config
+}: {
+  config?: FrontierClientAPIPlatformOptions;
+}) => {
+  const appName = config?.appName || DEFAULT_API_PLATFORM_APP_NAME;
+  return (
+    <EmptyState
+      icon={
+        <Image
+          // @ts-ignore
+          src={keyIcon}
+          alt="keyIcon"
+        />
+      }
+      heading="No service account"
+      subHeading={`Create a new account to use the APIs of ${appName}`}
+      primaryAction={
+        <Button
+          data-test-id="frontier-sdk-new-service-account-btn"
+          variant="secondary"
+        >
+          Create new service account
+        </Button>
+      }
+    />
+  );
+};
+
+const NoAccess = () => {
+  return (
+    <EmptyState
+      icon={<ExclamationTriangleIcon />}
+      heading="Restricted Access"
+      subHeading={`Admin access required, please reach out to your admin incase you want to generate a key.`}
+    />
+  );
+};
+
+const useAccess = (orgId?: string) => {
```

Comment:
As its a hook, should be move it to hooks folder?

Reply:
The access logic is local to the component. Once we make it generic and start using it in another place, then we can move it to a separate file.